src/lib/abrt_xmlrpc.c

```diff
@@ -70,6 +78,19 @@ struct abrt_xmlrpc *abrt_xmlrpc_new_client(const char *url, int ssl_verify)
 #else
     curl_parms.user_agent        = "abrt";
 #endif
+    if (api_key != NULL) {
+        /* Inject "Authorization" header right after the User-Agent header */
+
+        /* The User-Agent header normally has the following format:
+         * User-Agent: libreport/2.15.2 Xmlrpc-c/1.51.0 Curl/7.79.1
+         *
+         * We modify it here so it becomes 3 headers:
+         * User-Agent: libreport/2.15.2
+         * Authorization: Bearer <api-key>
+         * X-Libreport-Extra-User-Agent: Xmlrpc-c/1.51.0 Curl/7.79.1
+         */
+        curl_parms.user_agent = g_strdup_printf("%s\r\nAuthorization: Bearer %s\r\nX-Libreport-Extra-User-Agent:", curl_parms.user_agent, api_key);
```

Comment:
I don't think this string is ever freed, but it should be given that it contains sensitive information.